src/Matrix.jl

```diff
@@ -6789,6 +6789,17 @@ function Base.hvcat(rows::Tuple{Vararg{Int}}, A::MatrixElem{T}...) where T <: NC
   return M
 end
 
```

Comment:
How about also adding this, which is also from the same source:
```suggestion

#to bypass the vec(collect(M)) which copies twice
function Base.vec(M::Generic.Mat)
  return vec(M.entries)
end

```